airflow/contrib/operators/s3_delete_objects_operator.py

```diff
@@ -77,6 +77,9 @@ def __init__(
     def execute(self, context):
         s3_hook = S3Hook(aws_conn_id=self.aws_conn_id, verify=self.verify)
 
+        if isinstance(self.keys, list) and len(self.keys) == 0:
```

Comment:
This could probably be simplified to:

```suggestion
        if not self.keys:
```

Additionally move it above the `s3_hook = S3Hook()` creation -- that connects to the AWS servers, and there's no need to do that if we're going to short-circuit.